src/main/scala/cc/factorie/app/nlp/ner/StackedChainNer.scala

```diff
@@ -585,55 +585,10 @@ class StackedChainNer[L<:NerTag](labelDomain: CategoricalDomain[String],
   }
 }
 
-object StackedChainNer {
-  def initLexicons() : Unit = synchronized {
-    lexicon.iesl.Month.toString()
-    lexicon.iesl.Day.toString()
-
-    lexicon.iesl.PersonFirst.toString()
-    lexicon.iesl.PersonFirstHigh.toString()
-    lexicon.iesl.PersonFirstHighest.toString()
-    lexicon.iesl.PersonFirstMedium.toString()
-
-    lexicon.iesl.PersonLast.toString()
-    lexicon.iesl.PersonLastHigh.toString()
-    lexicon.iesl.PersonLastHighest.toString()
-    lexicon.iesl.PersonLastMedium.toString()
-
-    lexicon.iesl.PersonHonorific.toString()
-
-    lexicon.iesl.Company.toString()
-    lexicon.iesl.JobTitle.toString()
-    lexicon.iesl.OrgSuffix.toString()
-
-    lexicon.iesl.Country.toString()
-    lexicon.iesl.City.toString()
-    lexicon.iesl.PlaceSuffix.toString()
-    lexicon.iesl.USState.toString()
-    lexicon.iesl.Continents.toString()
-
-    lexicon.wikipedia.Person.toString()
-    lexicon.wikipedia.Event.toString()
-    lexicon.wikipedia.Location.toString()
-    lexicon.wikipedia.Organization.toString()
-    lexicon.wikipedia.ManMadeThing.toString()
-    lexicon.iesl.Demonym.toString()
-
-    lexicon.wikipedia.Book.toString()
-    lexicon.wikipedia.Business.toString()
-    lexicon.wikipedia.Film.toString()
-
-    lexicon.wikipedia.LocationAndRedirect.toString()
-    lexicon.wikipedia.PersonAndRedirect.toString()
-    lexicon.wikipedia.OrganizationAndRedirect.toString()
-  }
-}
-
```

Comment:
@eelstretching I've updated the scaladoc a bit to reflect both explicit and implicit use cases a little more clearly and the implementation to handle the magic a little more elegantly. I understand wariness at implicit 'spooky action at a distance', but I hope I've used it judiciously here, and I've made some updates to the code to ensure that it doesn't leak out anywhere and that more explicit usages are treated as first-class citizens. 

For the example that you posted if you wanted lexicons from the static set provided by factorie you would do:
```scala
import cc.factorie.app.nlp.lexicon
import cc.factorie.app.nlp.lexicon.Lexicon
import cc.factorie.util.ModelProvider

class SomeCleverModel(, countries:Lexicon, battles:Lexicon)

val countries = new iesl.Country()(someInputStream)
val battles = new wikipedia.Battle()(someOtherInputStream)
val model = new SomeCleverModel(, countries, battles)
```
If you were using some other dynamic lexicons you would do:
```scala
import cc.factorie.app.nlp.lexicon
import cc.factorie.app.nlp.lexicon.Lexicon
import cc.factorie.util.ModelProvider

class SomeCleverModel(, foo:Lexicon, bar:Lexicon)

val foo = new GenericLexicon("foo", someInputStream)
val bar = new GenericLexicon("bar", someOtherInputStream)
val model = new SomeCleverModel(, foo, bar)
```

In both these cases the lexicon will be loaded eagerly (since they are class instances rather than objects) and I think both of these usages are reasonably explicit and nice, let me know if you disagree. However, neither of these use cases are typical within the existing code. The typical use case (in the existing code) is: 'I want to train a model and I don't know what lexicons I need. Give me the ability to get them all, so I don't have to change three things every time I want to add one, but don't make me load them all into memory when I'm only using some of them.' We provide 110 lexicons statically and `cc.factorie.app.nlp.ner.StackedChainNer` uses ~30. I don't think it makes sense to provide them all as constructor arguments (or individually in any circumstances). I also don't think it makes sense to require StackedChainNer to load 110 lexicons to use 30, so the `StaticLexicons` class with its lazy loading is a compromise that exists to preserve legacy functionality (where there was a single classpath-based lexicon loading source) while adding flexibility (you give a root path where your lexicons are located) and to support the typical use case.